nix/haskell.nix

```diff
@@ -74,4 +75,31 @@ let
 in
 {
   inherit hsPkgs;
+
+  cabal-docspec-check = final.stdenv.mkDerivation {
+    name = "cabal-docspec-check";
+
+    src = fs.toSource {
+      root = ./..;
+      fileset = fs.unions [
+        (fs.fileFilter (f: f.hasExt "cabal" || f.hasExt "hs") ./..)
+      ];
+    };
+
+    nativeBuildInputs = [
+      final.fd
+      final.cabal-docspec
+      (hsPkgs.ghcWithPackages
+        (ps: [ ps.latex-svg-image ] ++ lib.filter (p: p ? components.library)
```

Comment:
It should also be possible to avoid mentioning `latex-svg-image` here (eg by making sure we also depend on test-only dependencies), but this doesn't seem like a priority.